metricbeat/module/kubernetes/container/container_test.go

```diff
@@ -34,49 +34,65 @@ import (
-	logger := logp.NewLogger("kubernetes.container")
+type ContainerTestSuite struct {
+	suite.Suite
+	MetricsRepo          *util.MetricsRepo
+	NodeName             string
+	Namespace            string
+	PodName              string
+	ContainerName        string
+	AnotherContainerName string
+	PodId                util.PodId
+	Logger               *logp.Logger
+	NodeMetrics          *util.NodeMetrics
+	ContainerMetrics     *util.ContainerMetrics
+}
 
-	f, err := os.Open(testFile)
-	assert.NoError(t, err, "cannot open test file "+testFile)
+func (s *ContainerTestSuite) SetupTest() {
+	s.MetricsRepo = util.NewMetricsRepo()
+	s.NodeName = "gke-beats-default-pool-a5b33e2e-hdww"
+	s.Namespace = "default"
+	s.PodName = "nginx-deployment-2303442956-pcqfc"
+	s.ContainerName = "nginx"
+	s.AnotherContainerName = "sidecar"
 
-	body, err := ioutil.ReadAll(f)
-	assert.NoError(t, err, "cannot read test file "+testFile)
+	s.PodId = util.NewPodId(s.Namespace, s.PodName)
 
-	metricsRepo := util.NewMetricsRepo()
+	s.Logger = logp.NewLogger("kubernetes.container")
 
-	nodeName := "gke-beats-default-pool-a5b33e2e-hdww"
+	s.NodeMetrics = util.NewNodeMetrics()
+	s.NodeMetrics.CoresAllocatable = util.NewFloat64Metric(2)
+	s.NodeMetrics.MemoryAllocatable = util.NewFloat64Metric(146227200)
 
-	nodeMetrics := util.NewNodeMetrics()
-	nodeMetrics.CoresAllocatable = util.NewFloat64Metric(2)
-	nodeMetrics.MemoryAllocatable = util.NewFloat64Metric(146227200)
-	addNodeMetric(metricsRepo, nodeName, nodeMetrics)
+	s.ContainerMetrics = util.NewContainerMetrics()
+	s.ContainerMetrics.MemoryLimit = util.NewFloat64Metric(14622720)
+}
 
-	namespace := "default"
-	podName := "nginx-deployment-2303442956-pcqfc"
-	podId := util.NewPodId(namespace, podName)
-	containerName := "nginx"
+func (s *ContainerTestSuite) ReadTestFile(testFile string) []byte {
+	f, err := os.Open(testFile)
+	s.NoError(err, "cannot open test file "+testFile)
 
-	containerMetrics := util.NewContainerMetrics()
-	containerMetrics.MemoryLimit = util.NewFloat64Metric(14622720)
-	addContainerMetric(metricsRepo, nodeName, podId, containerName, containerMetrics)
+	body, err := ioutil.ReadAll(f)
+	s.NoError(err, "cannot read test file "+testFile)
 
-	events, err := eventMapping(body, metricsRepo, logger)
-	assert.NoError(t, err, "error mapping "+testFile)
+	return body
+}
 
-	assert.Len(t, events, 1, "got wrong number of events")
+func (s *ContainerTestSuite) TestEventMapping() {
+	s.MetricsRepo.DeleteAllNodeStore()
 
-	testCases := map[string]interface{}{
+	s.addNodeMetric(s.NodeMetrics)
+	s.addContainerMetric(s.ContainerName, s.ContainerMetrics)
+
+	body := s.ReadTestFile(testFile)
+	events, err := eventMapping(body, s.MetricsRepo, s.Logger)
+
+	s.basicTests(events, err)
+
+	cpuMemoryTestCases := map[string]interface{}{
 		"cpu.usage.core.ns":   43959424,
 		"cpu.usage.nanocores": 11263994,
 
-		"logs.available.bytes": int64(98727014400),
-		"logs.capacity.bytes":  int64(101258067968),
-		"logs.used.bytes":      28672,
```

Comment:
it was moved to `basicTests`

Reply:
I didn't removed them entirely. I just moved into a function `basicTests` since they were not related to cpu/memory and in case we wanted to add more tests in the future. `pod_test.go` is a good example of why it makes sense to move those fields in a separate method